pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensible/data/BrokerLoadData.java

```diff
@@ -0,0 +1,324 @@
+    }
+
+    public double getMaxResourceUsageWithWeight(final double cpuWeight, final double memoryWeight,
+                                                final double directMemoryWeight, final double bandwidthInWeight,
+                                                final double bandwidthOutWeight) {
+        return max(cpu.percentUsage() * cpuWeight, memory.percentUsage() * memoryWeight,
+                directMemory.percentUsage() * directMemoryWeight, bandwidthIn.percentUsage() * bandwidthInWeight,
+                bandwidthOut.percentUsage() * bandwidthOutWeight) / 100;
+    }
+
+    public double getMaxResourceUsageWithWeightWithinLimit(final double cpuWeight, final double memoryWeight,
+                                                           final double directMemoryWeight,
+                                                           final double bandwidthInWeight,
+                                                           final double bandwidthOutWeight) {
+        return maxWithinLimit(100.0d,
+                cpu.percentUsage() * cpuWeight, memory.percentUsage() * memoryWeight,
+                directMemory.percentUsage() * directMemoryWeight, bandwidthIn.percentUsage() * bandwidthInWeight,
+                bandwidthOut.percentUsage() * bandwidthOutWeight) / 100;
+    }
+
+    private static double getNicSpeedBytesInSec(ServiceConfiguration conf) {
+        return conf.getLoadBalancerOverrideBrokerNicSpeedGbps().isPresent()
+                ? conf.getLoadBalancerOverrideBrokerNicSpeedGbps().get() * gigaBitToByte : -1.0;
+    }
+
+    synchronized ResourceUsage getMsgThroughputInUsage(double nicSpeedBytesInSec) {
+        if (msgThroughputInUsage.usage != msgThroughputIn) {
+            msgThroughputInUsage = new ResourceUsage(msgThroughputIn, nicSpeedBytesInSec);
+        }
+        return msgThroughputInUsage;
+    }
+
+    synchronized ResourceUsage getMsgThroughputOutUsage(double nicSpeedBytesInSec) {
+        if (msgThroughputOutUsage.usage != msgThroughputOut) {
+            msgThroughputOutUsage = new ResourceUsage(msgThroughputOut, nicSpeedBytesInSec);
+        }
+        return msgThroughputOutUsage;
+    }
+
+    public double getMaxResourceUsageWithExtendedNetworkSignal(ServiceConfiguration conf) {
+
+        double nicSpeedBytesInSec = getNicSpeedBytesInSec(conf);
+        return maxWithinLimit(100.0d,
+                cpu.percentUsage() * conf.getLoadBalancerCPUResourceWeight(),
+                memory.percentUsage() * conf.getLoadBalancerMemoryResourceWeight(),
+                directMemory.percentUsage() * conf.getLoadBalancerDirectMemoryResourceWeight(),
+                bandwidthIn.percentUsage() * conf.getLoadBalancerBandwithInResourceWeight(),
+                bandwidthOut.percentUsage() * conf.getLoadBalancerBandwithOutResourceWeight(),
+                getMsgThroughputInUsage(nicSpeedBytesInSec).percentUsage()
+                        * conf.getLoadBalancerBandwithInResourceWeight(),
```

Comment:
This change will require updates in BrokerConfiguration.

Reply:
Can you explain more about why we require updates in BrokerConfiguration?

Reply:
I see. Maybe we can add it when we need it.

Reply:
sure. we can add it later when needed.